### PR TITLE
RefferrdBy added

### PR DESCRIPTION
**Pull request by asogwachiuba**
What is the PR:
RefferredBy added for email signup, google signup and apple signup

### CODE REVIEW

File: lib/core/services/signup/signup_service.dart [hunk 1]
```diff
@@ -48,30 +53,43 @@ class SignUpServiceImpl extends SignUpService {
...
-        "fullName": fullName,
-        "email": email,
-        "phoneNumber": phoneNumber,
-        "password": password,
-        "dob": dob,
-        "type": profileType,
-        "profileType": profileType,
-        "receiveIbloovNewsletters": receiveNewsletters,
-      };
+      final request = (referredBy == null || referredBy.isEmpty)
```
Comment (divincoder) [inline lib/core/services/signup/signup_service.dart:68]: This is not necessary, null is fine for referredBy.
No?

File: lib/core/services/signup/signup_service.dart [hunk 2]
```diff
@@ -91,16 +109,25 @@ class SignUpServiceImpl extends SignUpService {
...
+    required String dob,
+    String? referredBy,
+  }) async {
     try {
-      final request = {
-        "token": token,
-        "profileType": profileType,
-        "dob": dob,
-      };
+      final request = (referredBy == null || referredBy.isEmpty)
```
Comment (divincoder) [inline lib/core/services/signup/signup_service.dart:119]: Please refer to the comment above

File: lib/core/services/signup/signup_service.dart [hunk 3]
```diff
@@ -137,16 +164,25 @@ class SignUpServiceImpl extends SignUpService {
...
```
Comment (divincoder) [inline lib/core/services/signup/signup_service.dart:174]: refer to the comment above

**Review by divincoder — CHANGES_REQUESTED** [review]

File: lib/core/services/signup/signup_service.dart (hunk 1 above)
Comment (asogwachiuba) [reply lib/core/services/signup/signup_service.dart:68]: I did this cause there is no need having a referredBy param if its null. Like if its not useful why pass it as an api param.

Comment (divincoder) [reply lib/core/services/signup/signup_service.dart:68]: There is no problem passing null to an API param. Was there an issue with the API when you passed null? 
when you call the endpoint with that value not existent, it is still null being passed implicitly.

Comment (divincoder) [reply lib/core/services/signup/signup_service.dart:68]: Just making sure the code is elegant. Currently, I don't like those checks like that

Comment (asogwachiuba) [reply lib/core/services/signup/signup_service.dart:68]: Okay. I was on the road back home, thats why it took me time to reply. 
What you said makes sense. not passing it implicitly means the api interprets the value of the param as null. I will make the update this night.

Comment (asogwachiuba) [reply lib/core/services/signup/signup_service.dart:68]: The update has been done.

**Review by divincoder — APPROVED** [review]

**Pull request merged**